File: 2021/day-01/2021-01-1.jl
include("../../libs/julia/n0s1-aoc.jl")

function count_deeper(data)
    lines = [parse(Int64, i) for i in split(data, "\n")]
    sum([1 for i in range(2,length(lines)) if (lines[i] - lines[i-1]) > 0])
end

println(count_deeper(readstring("sample-2021-01.txt")))
println(count_deeper(readstring("input-2021-01.txt")))

File: 2021/day-01/2021-01-1.py
#!/usr/bin/env python3
import sys
sys.path.append("../../libs/python")
import n0s1_aoc as n0s1

def count_deeper(data):
    lines = [int(i) for i in data[:-1].split("\n")]
    num = 0
    for i in range(1,len(lines)):
        if lines[i] - lines[i-1] > 0:
            num += 1
    return num

print(count_deeper(n0s1.file_to_string("sample-2021-01.txt")))
print(count_deeper(n0s1.file_to_string("input-2021-01.txt")))
